src/components/ValueInput/ValueInput.js

```diff
@@ -50,23 +50,24 @@ import { visibleActiveAccountAssetsWithBalanceSelector } from 'selectors/assets'
 import { activeAccountMappedCollectiblesSelector } from 'selectors/collectibles';
 
 import type { RootReducerState } from 'reducers/rootReducer';
-import type { Rates, Balances } from 'models/Asset';
-import type { Option, HorizontalOption } from 'models/Selector';
+import type { Rates, Balances, AssetOption } from 'models/Asset';
+import type { Collectible } from 'models/Collectible';
 import type { Theme } from 'models/Theme';
 import type { TransactionFeeInfo } from 'models/Transaction';
 
 import ValueInputHeader from './ValueInputHeader';
 
 export type ExternalProps = {
   disabled?: boolean,
-  customAssets?: Option[],
+  customAssets?: AssetOption[],
   customBalances?: Balances,
   selectorOptionsTitle?: string,
-  assetData: Option,
-  onAssetDataChange: (Option) => void,
+  assetData: AssetOption | Collectible,
+  onAssetDataChange: (AssetOption) => mixed,
+  // Called iff selected asset is Collectible
+  onCollectibleAssetDataChange?: (Collectible) => mixed,
```

Comment:
Looking at the props, it's not entirely clear if on collectible selection both methods are called, or only the second one, since collectibles here also count as assets.

Reply:
It is not called. Do you think a comment will be enough?

I struggle here with naming, as `Asset` type (and `AssetOption`) seem to hold only non-collectible assets. Is there any ubiquitous name used in our app for such assets?

Reply:
I'm not sure how to solve the naming -- maybe using "token asset" locally, like the type constants? The tabs on the assets screen are also "Tokens" and "Collectibles".

IMO it would be simpler to merge the change events into one and leave checking asset type to the handler, but if that is not possible then a comment that makes it clear which callbacks are called in both cases would be welcome.

Reply:
`Token` is also problematic as we have `tokenType` field in `Collectible` :-/

Reply:
I wanted to have separate events because they return two different types. While I wanted to avoid returning `AssetOption | Collectible` from a single method because majority use-cases do not support `Collectibles` and I didn't want to confuse the majority use case.

Reply:
I'll simply add comment then :-)